api/_docker/varnish/conf/default.vcl

```diff
@@ -99,18 +101,114 @@ sub vcl_recv {
     return (synth(400, "ApiPlatform-Ban-Regex HTTP header must be set."));
   }
 
+  # Allow purging (needs xkey mod)
+  if (req.method == "PURGE") {
+    if (!client.ip ~ invalidators) { # purge is the ACL defined at the begining
+      # Not from an allowed IP? Then die with an error.
+      return (synth(403, "This IP is not allowed to send PURGE requests."));
+    }
+
+    if (req.http.xkey) {
+      set req.http.n-gone = xkey.purge(req.http.xkey);
+      # or: set req.http.n-gone = xkey.softpurge(req.http.xkey)
+      return (synth(200, "Invalidated "+req.http.n-gone+" objects"));
+    }
+
+    # If you got this stage (and didn't error out above), purge the cached result
+    return (purge);
+  }
+
+  # Strip hash, server doesn't need it.
+  if (req.url ~ "\#") {
+    set req.url = regsub(req.url, "\#.*$", "");
+  }
+
+  # Strip a trailing ? if it exists
+  if (req.url ~ "\?$") {
+    set req.url = regsub(req.url, "\?$", "");
+  }
+
+  #if (req.http.Cache-Control ~ "(?i)no-cache") {
```

Comment:
Will this need to be completed before merging to properly allow for cache invalidation?

Reply:
Perhaps this is covered by line 105-119?